src/v/cluster/metadata_dissemination_types.h

```diff
@@ -38,6 +38,8 @@ struct ntp_leader : serde::envelope<ntp_leader, serde::version<0>> {
       , term(term)
       , leader_id(leader_id) {}
 
+    friend bool operator==(const ntp_leader&, const ntp_leader&) = default;
```

Comment:
nice find... how did come across this issue, just curious..

Reply:
do you mean how was the missing operator found? when inheriting from serde::envelope types will compare equal to each other--even if they aren't equal.

Reply:
> when inheriting from serde::envelope types will compare equal to each other--even if they aren't equal.

Right, I mean tests were always passing, so I was wondering what made you look at these missing operators and false positives in general.